bundles/org.openhab.binding.meteofrance/src/main/java/org/openhab/binding/meteofrance/internal/handler/MeteoFranceBridgeHandler.java

```diff
@@ -0,0 +1,172 @@
+    private static final String CARTE_VIGILANCE_URL = PORTAIL_API_BASE_URL.formatted("cartevigilance");
+
+    private static final String RAIN_FORECAST_BASE_URL = "https://rpcache-aa.meteofrance.com/internet2018client/2.0/nowcast/rain?lat=%.4f&lon=%.4f&token=__Wj7dVSTjV9YGu1guveLyDq0g7S7TfTjaHBTPTpO0kj8__";
+
+    private static final long CACHE_EXPIRY = TimeUnit.MINUTES.toMillis(10);
+
+    private final Logger logger = LoggerFactory.getLogger(MeteoFranceBridgeHandler.class);
+    private final Properties header = new Properties();
+    private final MeteoFranceDeserializer deserializer;
+
+    private final ExpiringCache<VigilanceEnCours> vigilanceText;
+    private final ExpiringCache<VigilanceEnCours> vigilanceMap;
+
+    public MeteoFranceBridgeHandler(Bridge bridge, MeteoFranceDeserializer deserializer) {
+        super(bridge);
+        this.deserializer = deserializer;
+
+        vigilanceText = new ExpiringCache<>(CACHE_EXPIRY, () -> this.getVigilanceEnCours(TEXTE_VIGILANCE_URL));
+        vigilanceMap = new ExpiringCache<>(CACHE_EXPIRY, () -> this.getVigilanceEnCours(CARTE_VIGILANCE_URL));
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Meteo-France API bridge handler.");
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        if (config.apikey.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.config-error-unknown-apikey");
+            return;
+        }
+        header.put("apikey", config.apikey);
+        header.put("accept", "*/*");
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> vigilanceText.getValue());
+    }
+
+    @Override
+    public void dispose() {
+        header.clear();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // The bridge does not handle commands
+    }
+
+    private @Nullable VigilanceEnCours getVigilanceEnCours(String url) {
+        try {
+            String answer = HttpUtil.executeUrl(HttpMethod.GET, url, header, null, null, REQUEST_TIMEOUT_MS);
+            logger.trace(answer);
+
+            VigilanceEnCours vigilance = deserializer.deserialize(MeteoFrance.VigilanceEnCours.class, answer);
+            if (vigilance.code() != 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, vigilance.message());
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+                return vigilance;
+            }
+        } catch (MeteoFranceException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Exception deserializing API answer: %s".formatted(e.getMessage()));
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
```

Comment:
I would expect an IO exception something like a timeout. If you expect a 401 (due to wrong auth key) it might be possible to setup some condition and to explicitly handel that case to a `CONFIGURATION_ERROR` ? 
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
```

Reply:
You're correct and `CONFIGURATION_ERROR` is handled above, this should be a `COMMUNICATION_ERROR`. Changed.